sql/src/main/java/org/opensearch/sql/sql/SQLService.java

```diff
@@ -75,6 +91,83 @@ private AbstractPlan plan(
                     .isExplain(request.isExplainRequest())
                     .build()));
 
+    // There is no full support for JSON format yet for in memory operations, aliases, literals, and casts
+    // Aggregation has differences with legacy results
+    if (request.format().getFormatName().equals("json")) {
+      class VisitorContext {
+        @Getter
+        @Setter
+        boolean isJSONSupported = true;
+      }
+
+      class JsonSupportedVisitor extends AbstractNodeVisitor<Boolean, VisitorContext> {
+        @Override
+        public Boolean visit(Node node, VisitorContext context) {
+          // A node is supported if all of its children are supported.
+          return node.getChild().stream().filter(c -> c.accept(this, context) != null)
+              .allMatch(c -> c.accept(this, context));
+        }
+
+        @Override
+        public Boolean visitProject(Project node, VisitorContext context) {
+          // Overridden visit functions are done in memory and are not supported with json format
+          class UnsupportedExpressionsVisitor
+              extends AbstractNodeVisitor<Boolean, Object> {
+
+            @Override
+            public Boolean visitFunction(Function node, Object context) {
+              // queries with function calls are not supported.
+              return false;
+            }
+
+            @Override
+            public Boolean visitLiteral(Literal node, Object context) {
+              // queries with literal values are not supported
+              return false;
+            }
+
+            @Override
+            public Boolean visitCast(Cast node, Object context) {
```

Comment:
I believe these unsupported function calls should be handled by the base class.

Reply:
Thanks, fixed in f6db3453e5b8aabdf10a6ba12016e311d0835c6b